File: 3rdWeek/특이한 정렬.js
function solution(numlist, n) {
    numlist.sort((a, b) => {
        if (Math.abs(a - n) > Math.abs(b - n)){
           return 1;
        }
        else if (Math.abs(a - n) < Math.abs(b - n)) {
            return -1;
        } else {
            if (a > b) return -1;
            else return 1;
        }
    })
   return numlist;
}

File: 3rdWeek/rookedsysc/특이한정렬.py
from functools import cmp_to_key

def solution(numlist, n):
    def custom_sort(a,b) :
        # b가 a보다 가까운 경우 정렬 필요함
        if abs(n - a) > abs(n - b) :
            return 1
        # b와 a의 거리가 같은데 b가 a 보다 더 큰 경우 정렬 필요함
        elif (abs(n - a) == abs(n - b)) and a < b :
            return 1
        return -1
    answer = sorted(numlist, key=cmp_to_key(custom_sort))
    return answer